### PR TITLE
Make lazy array load data JIT.

### PR DESCRIPTION
**Pull request by Cuiyansong**
Update lazy-array and add some tests.

### CODE REVIEW

File: addon/models/lazy-array.js [hunk 1]
```diff
@@ -10,70 +10,57 @@ export default Ember.ArrayProxy.extend({
...
+  chunkSize: undefined,
+
   init: function () {
-    var lazyContent = [];
     var totalCount = this.get('totalCount');
-    for (var i = 0; i < totalCount; i++) {
-      lazyContent.push(Ember.ObjectProxy.create({isLoaded: false}));
-    }
+    var lazyContent = new Array(totalCount);
+    lazyContent[0] = undefined; //hack: array[0] will be set to length of array in Chrome
```
Comment (jefflage) [inline addon/models/lazy-array.js]: Could you provide some context/writeup about how you found an issue here?

File: addon/models/lazy-array.js [hunk 2]
```diff
@@ -10,70 +10,57 @@ export default Ember.ArrayProxy.extend({
...
-    return loadedCount <= index + 11;
+  tryPreload: function (index, chunkSize) {
+    var chunkEndIndex = Math.floor(index / chunkSize + 1) * chunkSize - 1;
+    var lastRowIndex = this.get('totalCount') - 1;
+    if (chunkEndIndex >= lastRowIndex) { return; }
+
+    if (chunkEndIndex - index <= this.get('_preloadGate')) {
+      this.objectAt(chunkEndIndex + 1);
+    }
   },
```
Comment (jefflage) [inline addon/models/lazy-array.js]: What problem does this function solve?

Comment (Cuiyansong) [inline addon/models/lazy-array.js]: This function is going to try loading more data when we access the data which index near the 90% of lazy-array, and lazy-array make a property of "_preloadGate"

File: addon/models/lazy-array.js (hunk 1 above)
Comment (cyjia) [inline addon/models/lazy-array.js]: No need to hack anything, `totalCount` passed in is String type. Solved in #15 .

**Pull request merged**